src/Symfony/Component/Uuid/Uuid.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uuid;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Uuid implements \JsonSerializable
+{
+    public const TYPE_1 = UUID_TYPE_TIME;
+    public const TYPE_3 = UUID_TYPE_MD5;
+    public const TYPE_4 = UUID_TYPE_RANDOM;
+    public const TYPE_5 = UUID_TYPE_SHA1;
+
+    public const VARIANT_NCS = UUID_VARIANT_NCS;
+    public const VARIANT_DCE = UUID_VARIANT_DCE;
+    public const VARIANT_MICROSOFT = UUID_VARIANT_MICROSOFT;
+    public const VARIANT_OTHER = UUID_VARIANT_OTHER;
+
+    private $uuid;
+
+    public function __construct(string $uuid)
+    {
+        if (!uuid_is_valid($uuid)) {
+            throw new \InvalidArgumentException('The UUID is not valid.');
+        }
+
+        $this->uuid = $uuid;
+    }
+
+    public static function v1(): self
+    {
+        return new self(uuid_create(self::TYPE_1));
+    }
+
+    public static function v3(self $uuidNamespace, string $name): self
+    {
+        return new self(uuid_generate_md5($uuidNamespace->uuid, $name));
+    }
+
+    public static function v4(): self
+    {
+        return new self(uuid_create(self::TYPE_4));
+    }
+
+    public static function v5(self $uuidNamespace, string $name): self
+    {
+        return new self(uuid_generate_sha1($uuidNamespace->uuid, $name));
+    }
+
+    public static function fromBinary(string $uuidAsBinary): self
+    {
+        return new self(uuid_unparse($uuidAsBinary));
+    }
+
+    public static function isValid(string $uuid): bool
+    {
+        return uuid_is_valid($uuid);
+    }
+
+    public function toBinary(): string
+    {
+        return uuid_parse($this->uuid);
+    }
+
+    public function isNull(): bool
+    {
+        return uuid_is_null($this->uuid);
+    }
+
+    public function equals(self $other): bool
+    {
+        return $this->uuid === $other->uuid;
```

Comment:
`return 0 === uuid_compare($this->uuid, $other->uuid);` to accound for case differences
btw, the polyfill is bugged right now on this case insensitivity topic.